streaming/base/dataset.py

```diff
@@ -264,6 +265,10 @@ def __init__(self,
         for stream_id, stream in enumerate(self.streams):
             stream_shards = stream.get_shards(world)
             samples = sum(map(len, stream_shards))
+            if samples == 0:
+                index_json_filepath = os.path.join(stream.local, stream.split,
```

Comment:
I would call this `index_path` or so, in the eventuality that we make the path to the index.json an optional streaming dataset argument

Reply:
Done in https://github.com/mosaicml/streaming/pull/242/commits/004a43a9396d6b9b7a673015ca550d7d9cb2bffd.